packages/itwinui-react/src/core/LabeledInput/LabeledInput.tsx

```diff
@@ -104,25 +105,24 @@ export const LabeledInput = React.forwardRef((props, ref) => {
         </Label>
       )}
 
-      <InputWithDecorations
-        status={status}
-        isDisabled={disabled}
+      <Box
+        className={cx('iui-input-with-icon', inputWrapperProps?.className)}
         {...inputWrapperProps}
       >
-        <InputWithDecorations.Input
+        <Input
           disabled={disabled}
           required={required}
+          status={status}
           id={id}
           ref={ref}
           {...rest}
         />
         {shouldShowIcon && (
-          <Icon fill={!svgIcon ? status : undefined} padded {...iconProps}>
+          <EndIcon status={status} {...iconProps}>
```

Comment:
How do we support the case where user wants to override fill?

Reply:
Ah ok, it can be handled using `iconProps={{ fill: 'default' }}`.

Reply:
Since we are passing down `{...iconProps}` to `<EndIcon />`, and since `fill={status}` is above `{...rest}`, users can pass their custom fill via `iconProps` to override our fill.

```jsx
<LabeledInput
  label='label'
  message='message'
  status='negative'
  svgIcon={<SvgPlaceholder />}
  iconProps={{
    fill: 'hotpink',
  }}
/>
```

However, upon testing this, it doesn't work:

![image](https://github.com/iTwin/iTwinUI/assets/45748283/1de9f90c-d8b4-434e-8b5a-f470c0e017c3)

When testing how `<Icon>` behaves when `fill` is a custom string color, I realized that the fill is not applied as I expected ([Stackblitz playground](https://stackblitz.com/edit/github-psiuv8?file=src%2FApp.tsx)).

Is there an other way to pass custom fills to `<Icon>`?

Reply:
I'm not sure if we want to support custom fill. I was just asking about overriding the fill from a status color to `"default"`

Anything else could be achieved using CSS